File: JAVA/baju.java
import java.util.ArrayList;
import java.util.List;

public class baju extends aksesoris{
    private String size;
    private String merk;
    
        // CONSTRUCTOR
        public baju(){
            this.size = "";
            this.merk = "";
        }

        // CONSTRUCTOR dengan parameter
        public baju(String id, String nama, int harga, int stok, String jenis, String bahan, String warna, String size, String merk) {
            super(id, nama, harga, stok, jenis, bahan, warna); 
            this.size = size;
            this.merk = merk;
        }

        // Fungsi untuk menampilkan data dalam tabel
        public void tampilkan(List<baju> listProduk) {
            System.out.print("\n");
            // Menyimpan panjang maksimal dari setiap kolom
            int maxId = 2, maxNama = 4, maxHarga = 5, maxStok = 4;
            int maxJenis = 5, maxBahan = 5, maxWarna = 5, maxSize = 4, maxMerk = 4;

            // Cari panjang maksimal dari setiap kolom
            for (baju produk : listProduk) {
                maxId = Math.max(maxId, produk.get_id().length());
                maxNama = Math.max(maxNama, produk.get_nama_produk().length());
                maxHarga = Math.max(maxHarga, String.valueOf(produk.get_harga_produk()).length());
                maxStok = Math.max(maxStok, String.valueOf(produk.get_stok_produk()).length());
                maxJenis = Math.max(maxJenis, produk.get_jenis().length());
                maxBahan = Math.max(maxBahan, produk.get_bahan().length());
                maxWarna = Math.max(maxWarna, produk.get_warna().length());
                maxSize = Math.max(maxSize, produk.get_size().length());
                maxMerk = Math.max(maxMerk, produk.get_merk().length());
            }
            
            // HEADER
            System.out.print("+" + createRepeatedString('-', maxId + 2) 
                    + "+" + createRepeatedString('-', maxNama + 2)
                    + "+" + createRepeatedString('-', maxHarga + 2)
                    + "+" + createRepeatedString('-', maxStok + 2)
                    + "+" + createRepeatedString('-', maxJenis + 2)
                    + "+" + createRepeatedString('-', maxBahan + 2)
                    + "+" + createRepeatedString('-', maxWarna + 2)
                    + "+" + createRepeatedString('-', maxSize + 2)
                    + "+" + createRepeatedString('-', maxMerk + 2) + "+\n");

            System.out.print("| " + leftPad("ID", maxId)
                    + " | " + leftPad("Nama", maxNama)
                    + " | " + leftPad("Harga", maxHarga)
                    + " | " + leftPad("Stok", maxStok)
                    + " | " + leftPad("Jenis", maxJenis)
                    + " | " + leftPad("Bahan", maxBahan)
                    + " | " + leftPad("Warna", maxWarna)
                    + " | " + leftPad("Size", maxSize)
                    + " | " + leftPad("Merk", maxMerk)
                    + " |\n");

            System.out.print("+" + createRepeatedString('-', maxId + 2) 
                    + "+" + createRepeatedString('-', maxNama + 2)
                    + "+" + createRepeatedString('-', maxHarga + 2)
                    + "+" + createRepeatedString('-', maxStok + 2)
                    + "+" + createRepeatedString('-', maxJenis + 2)
                    + "+" + createRepeatedString('-', maxBahan + 2)
                    + "+" + createRepeatedString('-', maxWarna + 2)
                    + "+" + createRepeatedString('-', maxSize + 2)
                    + "+" + createRepeatedString('-', maxMerk + 2) + "+\n");

            // DATA
            for (baju produk : listProduk) {
                System.out.print("| " + leftPad(produk.get_id(), maxId)
                        + " | " + leftPad(produk.get_nama_produk(), maxNama)
                        + " | " + leftPad(String.valueOf(produk.get_harga_produk()), maxHarga)
                        + " | " + leftPad(String.valueOf(produk.get_stok_produk()), maxStok)
                        + " | " + leftPad(produk.get_jenis(), maxJenis)
                        + " | " + leftPad(produk.get_bahan(), maxBahan)
                        + " | " + leftPad(produk.get_warna(), maxWarna)
                        + " | " + leftPad(produk.get_size(), maxSize)
                        + " | " + leftPad(produk.get_merk(), maxMerk)
                        + " |\n");
            }

            System.out.print("+" + createRepeatedString('-', maxId + 2) 
                    + "+" + createRepeatedString('-', maxNama + 2)
                    + "+" + createRepeatedString('-', maxHarga + 2)
                    + "+" + createRepeatedString('-', maxStok + 2)
                    + "+" + createRepeatedString('-', maxJenis + 2)
                    + "+" + createRepeatedString('-', maxBahan + 2)
                    + "+" + createRepeatedString('-', maxWarna + 2)
                    + "+" + createRepeatedString('-', maxSize + 2)
                    + "+" + createRepeatedString('-', maxMerk + 2) + "+\n");
        }

        // Untuk membuat string berulang
        private static String createRepeatedString(char c, int length) {
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < length; i++) {
                sb.append(c);
            }
            return sb.toString();
        }

        // Untuk padding kiri
        private static String leftPad(String s, int width) {
            StringBuilder sb = new StringBuilder(s);
            while (sb.length() < width) {
                sb.append(' ');
            }
            return sb.toString();
        }

        // GETTER
        public String get_size(){
            return this.size; 
        }
        public String get_merk(){
            return this.merk; 
        }
        
        // SETTER
        public void set_size(String size){
            this.size = size;
        }
        public void set_merk(String merk){
            this.merk = merk;
        }
};

File: CPP/baju.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <string>
#include "aksesoris.cpp"

using namespace std;

class baju :public aksesoris
{
    private:
        string size;
        string merk;
    
        
    // CONSTRUCTOR
    public:
        baju(){
            this->size = "";
            this->merk = "";
        }
        
        baju(string id, string nama_produk, int harga_produk, int stok_produk, string jenis, string bahan, string warna, string size, string merk)
        : aksesoris(id, nama_produk, harga_produk, stok_produk, jenis, bahan, warna)
        {
            this->size = size;
            this->merk = merk;
        }

        // Fungsi menampilkan data dalam tabel
        void tampilkan(list<baju> list_produk) {
            // Menyimpan panjang maksimal dari setiap kolom
            size_t max_id = 2, max_nama = 4, max_harga = 5, max_stok = 4;
            size_t max_jenis = 5, max_bahan = 5, max_warna = 5, max_size = 4, max_merk = 4;
        
            // Cari panjang maksimal dari setiap kolom
            for (auto &produk : list_produk) { 
                max_id = max(max_id, produk.get_id().length());
                max_nama = max(max_nama, produk.get_nama_produk().length());
                max_harga = max(max_harga, to_string(produk.get_harga_produk()).length());
                max_stok = max(max_stok, to_string(produk.get_stok_produk()).length());
                max_jenis = max(max_jenis, produk.get_jenis().length());
                max_bahan = max(max_bahan, produk.get_bahan().length());
                max_warna = max(max_warna, produk.get_warna().length());
                max_size = max(max_size, produk.get_size().length());
                max_merk = max(max_merk, produk.get_merk().length());
            }
            
            // HEADER
            cout << "+" << string(max_id + 2, '-') 
                 << "+" << string(max_nama + 2, '-')
                 << "+" << string(max_harga + 2, '-')
                 << "+" << string(max_stok + 2, '-')
                 << "+" << string(max_jenis + 2, '-')
                 << "+" << string(max_bahan + 2, '-')
                 << "+" << string(max_warna + 2, '-')
                 << "+" << string(max_size + 2, '-')
                 << "+" << string(max_merk + 2, '-') << "+\n";
        
            cout << "| " << setw(max_id) << left << "ID"
                 << " | " << setw(max_nama) << left << "Nama"
                 << " | " << setw(max_harga) << left << "Harga"
                 << " | " << setw(max_stok) << left << "Stok"
                 << " | " << setw(max_jenis) << left << "Jenis"
                 << " | " << setw(max_bahan) << left << "Bahan"
                 << " | " << setw(max_warna) << left << "Warna"
                 << " | " << setw(max_size) << left << "Size"
                 << " | " << setw(max_merk) << left << "Merk"
                 << " |\n";
        
            cout << "+" << string(max_id + 2, '-') 
                 << "+" << string(max_nama + 2, '-')
                 << "+" << string(max_harga + 2, '-')
                 << "+" << string(max_stok + 2, '-')
                 << "+" << string(max_jenis + 2, '-')
                 << "+" << string(max_bahan + 2, '-')
                 << "+" << string(max_warna + 2, '-')
                 << "+" << string(max_size + 2, '-')
                 << "+" << string(max_merk + 2, '-') << "+\n";
        
            // DATA
            for (auto &produk : list_produk) {
                cout << "| " << setw(max_id) << left << produk.get_id()
                     << " | " << setw(max_nama) << left << produk.get_nama_produk()
                     << " | " << setw(max_harga) << left << produk.get_harga_produk()
                     << " | " << setw(max_stok) << left << produk.get_stok_produk()
                     << " | " << setw(max_jenis) << left << produk.get_jenis()
                     << " | " << setw(max_bahan) << left << produk.get_bahan()
                     << " | " << setw(max_warna) << left << produk.get_warna()
                     << " | " << setw(max_size) << left << produk.get_size()
                     << " | " << setw(max_merk) << left << produk.get_merk()
                     << " |\n";
            }
        
            cout << "+" << string(max_id + 2, '-') 
                 << "+" << string(max_nama + 2, '-')
                 << "+" << string(max_harga + 2, '-')
                 << "+" << string(max_stok + 2, '-')
                 << "+" << string(max_jenis + 2, '-')
                 << "+" << string(max_bahan + 2, '-')
                 << "+" << string(max_warna + 2, '-')
                 << "+" << string(max_size + 2, '-')
                 << "+" << string(max_merk + 2, '-') << "+\n";
        }
        // DESTRUCTOR
        ~baju(){
        }

    // GETTER
    string get_size(){
        return size; 
    }
    string get_merk(){
        return merk;
    }

    // SETTER
    void set_size(string size){
        this->size = size;
    }
    void set_merk(string merk){
        this->merk = merk;
    }
};
